### PR TITLE
feat: respect Media Display preferences

### LINKED ISSUE #1002: "Always expand content warnings" and "always show media" settings do not transfer and are not findable
### Pre-Checks

- [X] Check that there isn't [already an issue](https://github.com/elk-zone/elk/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/elk-zone/elk/discussions) or join our [Discord Chat Server](https://chat.elk.zone).
- [X] Providing a screenshot or video to reproduce the issue or show visually what was meant.
- [ ] I am willing to provide a PR.

### Describe the bug

I've never submitted a bug report here before. I copied this format from a couple other issues. Please be gentle if I messed up!

### Clear and concise description of the problem

I have the "Always expand posts marked with content warnings" and "Always show media" settings enabled on disabled.social.

When I logged into Elk, posts marked with content warnings and posts with just media marked as sensitive both stayed hidden, the default setting on Mastodon. I looked for a way to turn these settings back on and could not find it.

### Suggested solution

Make it so these settings transfer into Elk when someone logs in, like how my filters transferred just fine.

### Alternative

Add an easy-to-find way to re-enable these settings after logging into Elk.

### Additional context

If it's relevant, I'm on disabled.social, and I'm using Firefox on desktop. I've also attached a screenshot of one of the posts with a content warning that didn't expand, since the pre-check said to add a screenshot, though I'm not sure how helpful that is in this case.

Thank you for making this client!

### Reproduction video or screenshot

image

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by ayoayco**
fixes #1002 

This PR will respect two preferences with regards to media display: `show all`, or `hide all`
1. `show all` will expand all media hidden under a spoiler UI
2. `hide all` will put all media (even non-sensitive ones) under a spoiler UI

Also added a new translation string to mark if media is hidden, same with Mastodon web. Example below is a non-sensitive post but hidden to respect the "hide all" preference:
| Mastodon | Elk |
|--------|--------|
| Screenshot 2023-05-05 at 11 22 44 AM | Screenshot 2023-05-05 at 11 22 54 AM |

### CODE REVIEW

**Comment by ayoayco** [comment]
Ah sorry missed some need to fix typechecks...

**Comment by patak-dev** [comment]
ya, better to remove the lock file changes entirely for this PR

File: components/status/StatusContent.vue [hunk 1]
```diff
@@ -22,6 +22,12 @@ const isFiltered = $computed(() => status.account.id !== currentUser.value?.acco
 // check spoiler text or media attachment
 // needed to handle accounts that mark all their posts as sensitive
 const hasSpoilerOrSensitiveMedia = $computed(() => !!status.spoilerText || (status.sensitive && !!status.mediaAttachments.length))
+const unfilteredSensitive = !isFiltered && status.sensitive && !status.spoilerText
```
Comment (patak-dev) [inline components/status/StatusContent.vue:25]: This should be a computed

Comment (ayoayco) [reply components/status/StatusContent.vue]: pushed a change to make it computed

File: components/status/StatusContent.vue [hunk 2]
```diff
@@ -32,15 +38,15 @@ const hasSpoilerOrSensitiveMedia = $computed(() => !!status.spoilerText || (stat
...
+    <StatusBody v-if="unfilteredSensitive || hideAllMedia" :status="status" :newer="newer" :with-action="!isDetails" :class="isDetails ? 'text-xl' : ''" />
+    <StatusSpoiler :enabled="hasSpoilerOrSensitiveMedia || isFiltered" :filter="isFiltered" :unfiltered-sensitive="unfilteredSensitive || hideAllMedia" :is-d-m="isDM">
       <template v-if="status.spoilerText" #spoiler>
         <p>{{ status.spoilerText }}</p>
       </template>
       <template v-else-if="filterPhrase" #spoiler>
         <p>{{ `${$t('status.filter_hidden_phrase')}: ${filterPhrase}` }}</p>
       </template>
-      <StatusBody v-if="!status.sensitive || status.spoilerText" :status="status" :newer="newer" :with-action="!isDetails" :class="isDetails ? 'text-xl' : ''" />
+      <StatusBody v-if="!(unfilteredSensitive || hideAllMedia)" :status="status" :newer="newer" :with-action="!isDetails" :class="isDetails ? 'text-xl' : ''" />
```
Comment (patak-dev) [inline components/status/StatusContent.vue:49]: We weren't checking for `isFiltered` before here, is it ok to do it now?

Comment (ayoayco) [reply components/status/StatusContent.vue]: good catch, pushed a change for this.

Though I think for further improvement we need to have a different handling for hiding media. Maybe using the blurhash like in Mastodon. With what we have now, there's a complex scenario that Elk cannot support, which is if the post has a CW and a sensitive media, and the user viewing set "expand all CWs" but have "hide all media".

**Review by patak-dev — APPROVED** [review]

**Pull request merged**